File: PDP Second Project/src/Taktik.c
#include "Taktik.h"

Taktik TaktikOlustur()
{
    Taktik this;
	this = (Taktik)malloc(sizeof(struct TAKTIK));
	this->pop = 0;
	this->TaktikPopulasyonAl = &TaktikPopulasyonAl;
    this->TaktikYoket = &TaktikYoket;
	return this;
}

void TaktikPopulasyonAl(const Taktik this, int pop)
{
	this->pop = pop;
}

void TaktikYoket(const Taktik this){
	if(this == NULL) return;
	free(this);
}

File: PDP_3rd_Project/src/Taktik.java
public abstract class Taktik {
	protected int pop;
	
	public Taktik() {
		this.pop = 0;
	}
	public void TaktikPopulasyonAl(int pop) {
		this.setPop(pop);
	}
	public abstract int Savas();
	
	public void setPop(int pop) {
		this.pop = pop;
	}
}
